packages/grz-pydantic-models/src/grz_pydantic_models/submission/metadata/v1.py

```diff
@@ -276,13 +278,42 @@ class ResearchConsent(StrictBaseModel):
     Date of the delivery of the research consent in ISO 8601 format (YYYY-MM-DD)
     """
 
-    scope: object
+    # try Consent model first, falling back to dict
+    scope: Annotated[Consent | dict, Field(union_mode="left_to_right")]
     """
     Scope of the research consent in JSON format following the MII IG Consent v2025 FHIR schema. 
     See 'https://www.medizininformatik-initiative.de/Kerndatensatz/KDS_Consent_V2025/MII-IG-Modul-Consent.html' and 
     'https://packages2.fhir.org/packages/de.medizininformatikinitiative.kerndatensatz.consent'.
     """
 
+    @model_validator(mode="after")
+    def ensure_top_level_provision_deny(self):
+        if isinstance(self.scope, Consent):
+            if (top_level_provision := self.scope.provision) and (top_level_provision.type != ProvisionType.DENY):
+                raise ValueError(
+                    f"The root provision type must be deny, not {self.scope.provision.type}, "
+                    "since the profile follows an opt-in consent scheme. "
+                    "Explicit opt-in consents must be made via nested provisions."
+                )
+        elif self.scope:
+            # TODO print a traceback of the validation error for helping the user debug, without raising the actual error
+            log.warning(
+                "Could not recognize the object passed to researchConsents[].scope as a FHIR Consent profile. "
+                "See below for errors encountered during validation."
+            )
+            try:
+                Consent(**self.scope)
+                raise RuntimeError(
+                    "Pydantic correctly parsed research consent scope but failed to assign it to the correct type during model validation"
+                )
```

Comment:
is this still likely to happen with the `union_mode="left_to_right"` defined above?